### PR TITLE
[CRUD:Edit] Add exception message

### PR DESCRIPTION
**Pull request by sir-kain**
## Subject

From the `CRUDController:editAction()` add an error message before rendering the html page if no field is defined on the  `configureFormFields()` function

## Changelog

```markdown
### Added
- Added exception message if no field is defined on the  `configureFormFields()` function
```

### CODE REVIEW

File: src/Controller/CRUDController.php [hunk 1]
```diff
@@ -300,6 +300,7 @@ public function deleteAction($id)
      *
      * @param int|string|null $id
      *
+     * @throws \RuntimeException     If no editable attribut is defined
```
Comment (OskarStark) [inline src/Controller/CRUDController.php]: attribut**E**, but IMO `field` would better

File: src/Controller/CRUDController.php [hunk 2]
```diff
@@ -330,6 +331,10 @@ public function editAction($id = null)
         $this->admin->setSubject($existingObject);
         $objectId = $this->admin->getNormalizedIdentifier($existingObject);
 
+        if (!$this->admin->getFormTabs()) {
+            throw new \RuntimeException(sprintf('No editable field defined. Did you forget to implement the "configureFormFields" method?'));
```
Comment (kunicmarko20) [inline src/Controller/CRUDController.php]: This line is too long, could you please brake it.

File: src/Controller/CRUDController.php [hunk 3]
```diff
@@ -300,6 +300,7 @@ public function deleteAction($id)
      *
      * @param int|string|null $id
      *
+     * @throws \RuntimeException     If no editable field is defined
      * @throws NotFoundHttpException If the object does not exist
      * @throws AccessDeniedException If access is not granted
```
Comment (OskarStark) [inline src/Controller/CRUDController.php]: can you please move the `\RuntimeException` below the `AccessDeniedException`?

**Comment by kunicmarko20** [comment]
Please add a test for this case

**Comment by OskarStark** [comment]
Thanks for the test, but can you add one returning false 
, too?

**Comment by kunicmarko20** [comment]
Hey, @sir-kain this is a good solution, could you maybe apply it to create and show actions also? so we have consistency between actions. Thank you

**Comment by sir-kain** [comment]
@OskarStark done?

**Comment by sir-kain** [comment]
@kunicmarko20 Yes, I thought about that.
But in another PR?

**Comment by OskarStark** [comment]
Thank you @sir-kaine, would be great if you could add another PR 👍🏻

File: src/Controller/CRUDController.php [hunk 4]
```diff
@@ -330,6 +331,12 @@ public function editAction($id = null)
         $this->admin->setSubject($existingObject);
         $objectId = $this->admin->getNormalizedIdentifier($existingObject);
 
+        if (!$this->admin->getFormTabs()) {
```
Comment (stloyd) [inline src/Controller/CRUDController.php]: This is BC break cause at this point form can be **not** initialized yet, but moving call:
```
$form = $this->admin->getForm();
```
Before this `if` fixes that exception.

ps. mine admin controller don't overwrite any method of `CRUDController` just adds new like in docs.

Comment (kunicmarko20) [reply src/Controller/CRUDController.php]: This was fixed in https://github.com/sonata-project/SonataAdminBundle/pull/5195

Comment (stloyd) [reply src/Controller/CRUDController.php]: Ah, just noticed fix: https://github.com/sonata-project/SonataAdminBundle/pull/5195

**Pull request merged**